src/wallet/wallet.cpp

```diff
@@ -1714,6 +1714,9 @@ CBlockIndex* CWallet::ScanForWalletTransactions(CBlockIndex* pindexStart, CBlock
 
     CBlockIndex* pindex = pindexStart;
     CBlockIndex* ret = nullptr;
+
+    if (pindex) LogPrintf("Rescan started from block %d...\n", pindex->nHeight);
```

Comment:
If !pindex, doesn't that mean `Rescan started from block 0...`?

Reply:
No, if `!pindex` the loop below won't run.

Reply:
Nit, could use `pindexStart` instead so it's more obvious?

Reply:
Nah it's fine, I was just confused then, it looked a bit strange to conditionally log this message.

Reply:
Indeed. I just want to prevent a nullptr access in case of stupid future changes (not possible now).

Reply:
Nit: I'd be down for an early return based on `pIndexStart` being null, given in that case all else is irrelevant.